src/api/schema/events/metric.rs

```diff
@@ -111,6 +115,17 @@ impl Metric {
                 .expect("JSON serialization of metric event failed. Please report."),
             EventEncodingType::Yaml => serde_yaml::to_string(&self.event)
                 .expect("YAML serialization of metric event failed. Please report."),
+            EventEncodingType::Logfmt => {
+                let json = serde_json::to_value(&self.event)
+                    .expect("logfmt serialization of metric event failed. Please report.");
+                match json {
+                    Value::Object(map) => encode_logfmt::to_string(
+                        &map.into_iter().collect::<BTreeMap<String, Value>>(),
+                    )
+                    .expect("logfmt serialization of metric event failed. Please report."),
+                    _ => panic!("logfmt serialization of metric event failed. Please report."),
```

Comment:
Very minor nit: Since these three errors are identical, I would either:

- Create a specific error for each, so it's clearerwhere serialization failed, or...
- Create a static string, and reference it in all three cases so we only need to edit it once.